sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -0,0 +1,359 @@
+			ret = -ENOMEM;
+			goto err;
+		}
+	}
+	dev_info(dev, "[ADSP] shared-dram vbase=%p, phy addr :%llx,  size=0x%x\n",
+		 adsp->shared_dram, adsp->pa_shared_dram, shared_size);
+err:
+	return ret;
+}
+
+static int mt8195_dsp_probe(struct snd_sof_dev *sdev)
+{
+	struct platform_device *pdev = container_of(sdev->dev, struct platform_device, dev);
+	struct adsp_priv *priv;
+	int ret, mailbox_type;
+
+	priv = devm_kzalloc(&pdev->dev, sizeof(*priv), GFP_KERNEL);
+	if (!priv)
+		return -ENOMEM;
+
+	sdev->pdata->hw_pdata = priv;
+	priv->dev = sdev->dev;
+	priv->sdev = sdev;
+
+	priv->adsp = devm_kzalloc(&pdev->dev, sizeof(struct adsp_chip_info),
+				  GFP_KERNEL);
+
+	if (!priv->adsp)
+		return -ENOMEM;
+
+	adsp_info = priv->adsp;
+
+	ret = platform_parse_resource(pdev, adsp_info);
+	if (ret)
+		return ret;
+
+	ret = platform_parse_clock(&pdev->dev);
+	if (ret)
+		return ret;
+
+	pm_runtime_enable(&pdev->dev);
+	pm_runtime_get_sync(&pdev->dev);
+
+	ret = adsp_clock_on(&pdev->dev);
+	if (ret) {
+		dev_err(sdev->dev, "adsp_clock_on fail!\n");
+		goto err_pm_disable;
+	}
+
+	ret = adsp_sram_power_on(sdev->dev, true);
+	if (ret) {
+		dev_err(sdev->dev, "adsp_sram_power_on fail!\n");
+		goto err_adsp_clock_off;
+	}
+
+	ret = adsp_memory_remap_init(&pdev->dev);
+	if (ret) {
+		dev_err(sdev->dev, "adsp_memory_remap_init fail!\n");
+		goto err_adsp_sram_power_off;
+	}
+
+	sdev->bar[SOF_FW_BLK_TYPE_IRAM] = devm_ioremap(sdev->dev,
+						       priv->adsp->pa_sram,
+						       priv->adsp->srammsize);
+	if (!sdev->bar[SOF_FW_BLK_TYPE_IRAM]) {
+		dev_err(sdev->dev, "failed to ioremap base %pa size 0x%x\n",
+			&priv->adsp->pa_sram, priv->adsp->srammsize);
+		ret = -EINVAL;
+		goto err_adsp_sram_power_off;
+	}
+
+	sdev->bar[SOF_FW_BLK_TYPE_SRAM] = devm_ioremap_wc(sdev->dev,
+							  priv->adsp->pa_dram,
+							  priv->adsp->dramsize);
+	if (!sdev->bar[SOF_FW_BLK_TYPE_SRAM]) {
+		dev_err(sdev->dev, "failed to ioremap base %pa size 0x%x\n",
+			&priv->adsp->pa_dram, priv->adsp->dramsize);
+		ret = -EINVAL;
+		goto err_adsp_sram_power_off;
+	}
+	adsp_info->va_dram = sdev->bar[SOF_FW_BLK_TYPE_SRAM];
+	mailbox_type = SOF_FW_BLK_TYPE_SRAM;
+
+	ret =  adsp_shared_base_ioremap(pdev, adsp_info);
+	if (ret) {
+		dev_err(sdev->dev, "adsp_shared_base_ioremap fail!\n");
+		goto err_adsp_sram_power_off;
+	}
+
+	sdev->mmio_bar = mailbox_type;
+	sdev->mailbox_bar = mailbox_type;
+
+	return 0;
+
+err_adsp_sram_power_off:
+	adsp_sram_power_on(&pdev->dev, false);
+err_adsp_clock_off:
+	adsp_clock_off(&pdev->dev);
+err_pm_disable:
+	pm_runtime_disable(&pdev->dev);
+
+	return ret;
+}
+
+static int mt8195_dsp_remove(struct snd_sof_dev *sdev)
+{
+	struct platform_device *pdev = container_of(sdev->dev, struct platform_device, dev);
+
+	adsp_clock_off(&pdev->dev);
+	adsp_sram_power_on(&pdev->dev, false);
+	pm_runtime_disable(&pdev->dev);
+	return 0;
+}
+
+static int mt8195_get_bar_index(struct snd_sof_dev *sdev, u32 type)
+{
+	return type;
+}
```

Comment:
is this a TODO? why return an argument as is?

Reply:
It is one-to-one match between type and Bar index so it just return type. thanks

Reply:
can you please add a comment to say this?

Reply:
ok. I will add comment for it. thanks